src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/InvokeRestMethodCommand.Common.cs

```diff
@@ -169,36 +170,43 @@ private bool TryConvertToXml(string xml, out object doc, ref Exception exRef)
 
         private bool TryConvertToJson(string json, out object obj, ref Exception exRef)
         {
-            bool converted;
+            bool converted = false;
             try
             {
                 ErrorRecord error;
                 obj = JsonObject.ConvertFromJson(json, out error);
-                converted = true;
 
                 if (null == obj)
                 {
-                    string pattern = @"^\s*null\s*$";
-                    Match matches = Regex.Match(json, pattern);
-                    converted = matches.Success;
+                    // This ensures that a null returned by ConvertFromJson() is the actual JSON null literal. 
+                    // if not, the ArgumentException will be caught.
+                    JToken.Parse(json);
```

Comment:
After I see the change I'm even more confident that we should make this check in [ConvertFromJson](https://github.com/PowerShell/PowerShell/blob/4bc52d2358222084738157a08907fac32d13bd3a/src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs#L71) - we should remove `if ((Regex.Match(input, @"^\s*\[")).Success)` and replace `JArray.Parse(input);` with `JToken.Parse(json);`. So we'll be closer to JavaScriptSerializer, and we can make it easier to move on one in the future. Also we exclude the same catch on level above.

I review a history of the code and discover that I see ways to clean up and refactor the code. So I agree to merge the PR as is and ready to refactor in days.

Reply:
Awesome. Just note that just like the change here, making that change `ConvertFromJson()` will be a breaking change since empty strings and whitespace-only strings will produce errors where as before they returned `$null`. I think that's appropriate, but the world at large may not. But you are right, we don't have much time. Will you handle the `ConvertFromJson()` refactor?

Reply:
I'd refactor the code in any case because it provokes troubles.

As for a breaking change I have the thoughts - should we fix the Issue now or leave the behavior as in Windows PowerShell? Will MSFT fix the Issue in Windows PowerShell too and when? Will MSFT fix the Issue in .Net JavaScriptSerializer and when? Should we request the fix in Newton.Json as we did for JArray?
I think to get sync fixes is better then a breaking change in only PowerShell Core.

Reply:
@iSazonov Good questions, but we should track that in a separate issue.